### ISSUE TITLE
Hover background color described in theme is not applied to input

### ISSUE THREAD

**Issue by ezze** [GeekyAnts/NativeBase#4444]
**Describe the bug**

`backgroundColor` specified for `_hover` state of input in a theme is not applied (in contrast to `borderColor` or `backgroundColor` of `_focus` state). It's also worth noting that `backgroundColor` for `_hover` works for inputs with `InputLeftElement` and/or `InputRightElement`.

**Steps to reproduce**

*src/theme/index.ts*

```typescript
import { extendTheme } from "native-base";

const theme = extendTheme({
  components: {
    Input: {
      baseStyle: {
        borderColor: "#000",
        backgroundColor: "#eee",
        _focus: {
          borderColor: "#00b",
          backgroundColor: "#fff"
        },
        _hover: {
          borderColor: "#00b",
          backgroundColor: "#fff"
        }
      }
    }
  }
});

export type ITheme = typeof theme;

export default theme;
```

*src/components/Example.tsx*

```typescript
import React from "react";
import {
  VStack,
  Text,
  Input
} from "native-base";

function Example() {
  return (
    <VStack space={4} width={500}>
      <VStack space={2}>
        <Text>Hover background from theme doesn't work for input</Text>
        <Input />
      </VStack>
      <VStack space={2}>
        <Text>Hover background from theme works for input with input left/right element</Text>
        <Input InputLeftElement={
          <Text>left</Text>
        } />
        <Input InputRightElement={
          <Text>right</Text>
        } />
      </VStack>
    </VStack>
  );
}

export default Example;
```

**Expected behaviour**

`backgroundColor` specified for `_hover` state of input without additional left/right elements in a theme is applied.

**CodeSandBox/Snack link**
[Here is a live example](https://codesandbox.io/s/confident-jepsen-us0tf?file=/src/components/Example.tsx)

**Platform**

- NativeBase version 3.2.2
- Browser Chromium 
- Browser version 96.0.4664.110 (Official Build) snap (64-bit)
- React Native version 0.64.2
- node 14.15.1
- yarn 1.22.17

**Comment by MD-REHMAN** [comment]
Thanks, @ezze for pointing it out. We have known about it and we're working on a fix.
For now instead of using `backgroundColor`, please use `bg`. And I'm very sure I'll fix the issue.

And I would recommend you to go through [this](https://md-rehman.hashnode.dev/customising-components-in-nativebase) blog for better understanding of how theme customisation works in NativeBases

**Comment by ezze** [comment]
@MD-REHMAN Sorry, `bg` also doesn't work in `_hover` for me. I have the same behavior. It works for inputs with addons (left/right elements) but doesn't work for inputs without addons:

**Without addons**

![image](https://user-images.githubusercontent.com/665479/150839536-9e28c120-4a56-47a9-8d76-0be1db9db9a2.png)

**With addons**

![image](https://user-images.githubusercontent.com/665479/150839393-aac2c553-798b-4b36-9bee-2f4d504c0dfc.png)

It also works for `_focus`:

file:///home/ezze/Desktop/Peek 2022-01-24 21-10.gif

**Comment by ankit-tailor** [comment]
Hi @ezze , 

Styles defined in `variants` will overwrite `baseStyle`. So we have defined default styling for `variants` and that is overwriting the `baseStyle` of `_hover`. So you can define a style for `variants` as shown in the below snippet. 

```
const theme = extendTheme({
  components: {
    Input: {
      variants: {
        outline: {
          _hover: {
            bg: 'teal.600',
          },
        },
      },
      baseStyle: {
        borderColor: '#000',
        bg: 'yellow.500',
        _focus: {
          borderColor: '#00b',
          bg: 'red.500',
        },
        _disabled: {
          bg: 'amber.500',
        },
      },
    },
  },
});
```
You can read more about customizing theme [here](https://md-rehman.hashnode.dev/customising-components-in-nativebase).

**Comment by ezze** [comment]
Is it really fixed?

**Comment by surajahmed** [comment]
@ezze We've fixed this issue in version `3.4.0`. Can you please try the latest version? 
Please feel free to reopen if this issue persists. Thanks.

**Comment by minnam** [comment]
I'm still having the same issue on 3.4.13

`
    variants: {
        secondary: (colorScheme) => {
          return {
            bg: 'grey.100', // works
            rounded: 'md', // works
            borderWidth: 1, // works
            borderColor: 'red.300', // not working
            _focus: {
              borderWidth: 1, // works
            },
          }
        },
      },
`
